keras/callbacks.py

```diff
@@ -590,12 +590,15 @@ class TensorBoard(Callback):
             write_graph is set to True.
         write_images: whether to write model weights to visualize as
             image in Tensorboard.
+        batch_size_histogram: size of batch of inputs to feed to the 
+            network for histograms computation.
     """
 
     def __init__(self, log_dir='./logs',
                  histogram_freq=0,
                  write_graph=True,
-                 write_images=False):
+                 write_images=False,
+                 batch_size_histogram=1):
```

Comment:
This can also be `None` by default, in this case `model.fit()` can pass batch_size to callback in `def _fit_loop()`.